### DISCUSSION TITLE
NT-Xent loss with multiple positive pairs

### DISCUSSION THREAD

**Discussion by annaVettoruzzo** [KevinMusgrave/pytorch-metric-learning#606] [Q&A]
Does NTXent loss address the case with multiple positive pairs? If the label assignment has the same label for multiple samples, does this loss sum over all possible positive pairs (similarly to Ed 2 and 3 in [https://arxiv.org/abs/2201.11736])? 

Thank you.

**Comment by KevinMusgrave** [comment]
Yes it will sum over all positive pairs, and the negative pairs for each positive pair will be based on the _anchor_ of the positive pair. 

For example, if the indices for a positive pair are (2, 5), then all of the negative pairs for this positive pair will be of the form (2, X).  

So I think the implementation is the same as Equation 2 in the paper you linked.

See these related comments: 
- https://github.com/KevinMusgrave/pytorch-metric-learning/issues/600#issuecomment-1483894037
- https://github.com/KevinMusgrave/pytorch-metric-learning/issues/6#issuecomment-587331886
